### PR TITLE
Update: source-loader lib migrated to use modern build tool

### PR DESCRIPTION
**Pull request by shariqx5**
Issue:

Reference: [#18732](https://github.com/storybookjs/storybook/issues/18732)

## What I did
Use tsup to build @storybook/source-loader
## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by shariqx5** [comment]
#18732

**Comment by shariqx5** [comment]
@ndelangen & @IanVS, Would you mind reviewing this PR? I have fixed most of the failing tests, but still, a few are failing.

**Review by IanVS — APPROVED** [review]
Looks good to me, thanks!

**Comment by shariqx5** [comment]
Unable to merge the branch, it says **You are not authorized to merge**. What could be the issue @IanVS ?

File: code/lib/source-loader/package.json [hunk 1]
```diff
@@ -51,5 +59,10 @@
   "publishConfig": {
     "access": "public"
   },
+  "bundler": {
+    "entries": [
+      "./src/index.ts"
+    ]
```
Comment (ndelangen) [inline code/lib/source-loader/package.json:65]:
```suggestion
    ],
    "platform": "node"
```

Comment (IanVS) [reply code/lib/source-loader/package.json]: Just curious, why was this needed?

**Review by ndelangen — APPROVED** [review]

File: code/lib/source-loader/package.json (hunk 1 above)
Comment (IanVS) [reply code/lib/source-loader/package.json]: I see you resolved this without answering, but I'd still love to understand so I can know whether it's needed in the future. Normally I've only added it when there's an error, but should we be adding it for any package that we expect to only run in node? Is that why it's added here?

Comment (ndelangen) [reply code/lib/source-loader/package.json]: source loader is node only. This ensures that even the ESM output is compiled for executing in node.

**Comment by shariqx5** [comment]
I'm not authorized to merge this 😞 Can you give me permission @ndelangen?

**Comment by ndelangen** [comment]
@shariqx5 you have an invite to the org 🎊

**Comment by ndelangen** [comment]
Do we have this addon used anywhere still? I'd like to check if it actually still functions. I'm not aware of any tests that cover this package.

**Pull request merged**